test/Graphics/RenderTarget.test.cpp

```diff
@@ -8,6 +8,12 @@
 
 class RenderTarget : public sf::RenderTarget
 {
+public:
+    RenderTarget() // NOLINT(modernize-use-equals-default)
```

Comment:
Idk why but using ` = default` breaks compilation. Maybe it's related to me moving some `= default`s from .cpp files to .hpp files.

```
/Users/thrasher/Projects/sfml/test/Graphics/RenderTarget.test.cpp:34:28: error: default initialization of an object of const type 'const RenderTarget' without a user-provided default constructor
        const RenderTarget renderTarget;
```

Reply:
This is an interesting error. The full error message with GCC 12.3 tells that it refuses to instantiate a const RenderTarget because:
> the implicitly-defined constructor does not initialize ‘bool sf::RenderTarget::StatesCache::enable’

Maybe `sf::RenderTarget::m_cache`  member variable should be initialized in `sf::RenderTarget` to fix the error instead of avoiding it?

Reply:
Good catch, that totally fixed it!